src/components/ui/Drawers/BaseDrawer.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * This file defines the base Drawer component.
+ * When an actual Drawer component is needed,
+ * it's necessary to create separate component files
+ * for each specific use, inheriting from the components in this file.
+ *
+ * This base Drawer component file defines only the minimal styles
+ * that are common across all drawers in the project.
+ *
+ * This file is based on the Sheet component from shadcn and customized for our needs.
+ * The Sheet component of shadcn is based on the Dialog component of Radix UI.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/sheet
+ * Radix UI: https://www.radix-ui.com/themes/docs/components/dialog
+ */
+'use client';
+
+import { cn } from '@/lib/tailwind/utils';
+
+import {
+  Close as PrimitiveClose,
+  Content as PrimitiveContent,
+  // Description as PrimitiveDescription,
+  Overlay as PrimitiveOverlay,
+  Portal as PrimitivePortal,
+  Root as PrimitiveRoot,
+  Title as PrimitiveTitle,
+  Trigger as PrimitiveTrigger,
+} from '@radix-ui/react-dialog';
+import { cva, type VariantProps } from 'class-variance-authority';
+import { ComponentPropsWithoutRef, ElementRef, forwardRef, HTMLAttributes } from 'react';
+
+const BaseDrawerRoot = PrimitiveRoot;
+const BaseDrawerTrigger = PrimitiveTrigger;
+const BaseDrawerClose = PrimitiveClose;
+const BaseDrawerPortal = PrimitivePortal;
+
+const BaseDrawerOverlay = forwardRef<
+  ElementRef<typeof PrimitiveOverlay>,
+  ComponentPropsWithoutRef<typeof PrimitiveOverlay>
+>(({ className, ...props }, ref) => (
+  <PrimitiveOverlay
+    className={cn(
+      'fixed inset-0 z-50',
+      'bg-overlay',
+      'data-[state=open]:animate-fadeIn data-[state=closed]:animate-fadeOut',
+      className,
+    )}
+    {...props}
+    ref={ref}
+  />
+));
+BaseDrawerOverlay.displayName = 'BaseDrawerOverlay';
+
+const BaseDrawerVariants = cva(cn('fixed z-50 bg-white flex flex-col'), {
+  variants: {
+    side: {
+      bottom: cn(
+        'inset-x-0 bottom-0 rounded-t max-h-[calc(100vh-2rem)]',
+        'data-[state=open]:animate-slideInFromBottom data-[state=closed]:animate-slideOutToBottom',
+      ),
+      right: cn(
+        'inset-y-0 right-0 h-full w-5/6 max-w-sm',
+        'data-[state=open]:animate-slideInFromRight data-[state=closed]:animate-slideOutToRight',
+      ),
+    },
```

Comment:
I decided not to add a `color` or `variant` property, which would look like the following code because the CSS `border-color` property is not inherited by children elements by the CSS3 default feature.

```tsx
variant: {
  white: 'bg-white border-gray-light',
  primary: 'bg-primary-lightest border-primary-light',
},
```